src/test/java/org/magicdgs/readtools/cmd/argumentcollections/FixBarcodeAbstractArgumentCollectionTest.java

```diff
@@ -0,0 +1,251 @@
+                        true},
+                // two quality tags and only one barcode quality (default)
+                {new ArgumentsBuilder()
+                        .addArgument("rawBarcodeQualityTag", "Q1")
+                        .addArgument("rawBarcodeQualityTag", "Q2"),
+                        true},
+                // two quality tags and only one barcode quality (overridden)
+                {new ArgumentsBuilder()
+                        .addArgument("rawBarcodeQualityTag", "Q1")
+                        .addArgument("rawBarcodeQualityTag", "Q2")
+                        .addArgument("rawBarcodeSequenceTags", "B1"),
+                        true},
+                // two barcode tags and only one quality (overridden)
+                {new ArgumentsBuilder()
+                        .addArgument("rawBarcodeSequenceTags", "B1")
+                        .addArgument("rawBarcodeSequenceTags", "B2")
+                        .addArgument("rawBarcodeQualityTag", "Q1"),
+                        true},
+                // incompatible arguments
+                {new ArgumentsBuilder()
+                        .addBooleanArgument("barcodeInReadName", true)
+                        .addArgument("rawBarcodeSequenceTags", "B1"),
+                        true},
+                // invalid qual tag names
+                {new ArgumentsBuilder()
+                        .addArgument("rawBarcodeQualityTag", "1A"),
+                        true},
+                {new ArgumentsBuilder()
+                        .addArgument("rawBarcodeQualityTag", "A1A"),
+                        true},
+                {new ArgumentsBuilder()
+                        .addArgument("rawBarcodeQualityTag", "A+"),
+                        true},
+                {new ArgumentsBuilder()
+                        .addArgument("rawBarcodeQualityTag", "+A"),
+                        true},
+                {new ArgumentsBuilder()
+                        .addArgument("rawBarcodeQualityTag", "1A"),
+                        true},
+                {new ArgumentsBuilder()
+                        .addArgument("rawBarcodeQualityTag", "A1A"),
+                        true},
+                {new ArgumentsBuilder()
+                        .addArgument("rawBarcodeQualityTag", "A+"),
+                        true},
+                {new ArgumentsBuilder()
+                        .addArgument("rawBarcodeQualityTag", "+A"),
+                        true}
+        };
+    }
+
+    @Test(dataProvider = "badArgs", expectedExceptions = CommandLineException.class)
+    public void testBadArguments(final ArgumentsBuilder args, final boolean fixQuals)
+            throws Exception {
+        args.addArgument("verbosity", "ERROR").addBooleanArgument("QUIET", true);
+        final CommandLineProgram clp = new FixBarcodesCLP(fixQuals);
+        clp.instanceMain(args.getArgsArray());
+    }
+
+    @DataProvider(name = "goodArgs")
+    public Object[][] getGoodArguments() {
```

Comment:
Won't fix, because this requires also the expected class and that's more difficult.